compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -80,9 +80,9 @@ object ProtoTypes {
                     val excluded = ctx.typerState.ownedVars.filter(!_.isInstantiated)
                     val aboveOK = !ctx.typerState.constraint.dependsOn(tvar, excluded, co = true)
                     val belowOK = !ctx.typerState.constraint.dependsOn(tvar, excluded, co = false)
-                    if aboveOK then
+                    if belowOK then
                       tvar.instantiate(fromBelow = true)
-                    else if belowOK then
+                    else if aboveOK then
                       tvar.instantiate(fromBelow = false)
```

Comment:
To clarify what I meant in my previous comment, the code after this commit is:
```scala
if belowOK then
  tvar.instantiate(fromBelow = true)
else if aboveOK then
  tvar.instantiate(fromBelow = false)
```
but if I swap the order of the branches:
```scala
if aboveOK then
  tvar.instantiate(fromBelow = false)
else if belowOK then
  tvar.instantiate(fromBelow = true)
```
then tests/pos/i21981.scala starts failing, so that must mean that aboveOK being true is not a good enough condition to ensure that instantiating a type variable is safe.